### PR TITLE
ref(ui): Make page time range selector reusable

### PR DESCRIPTION
**Pull request by matejminar**
image

Moving the PageTimeRangeSelector from org stats to a reusable component.
I'm going to need the same thing on the release details page.
No change in functionality.

### CODE REVIEW

File: static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    p.isCalendarOpen
+      ? `${p.theme.borderRadius} ${p.theme.borderRadius} 0 0`
+      : p.theme.borderRadius};
+  padding: 0;
+  margin: 0;
+  font-size: ${p => p.theme.fontSizeMedium};
+  color: ${p => p.theme.textColor};
+  z-index: ${p => p.theme.zIndex.globalSelectionHeader};
+
+  /* TimeRageRoot in TimeRangeSelector */
```
Comment (markstory) [inline static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx:54]:
```suggestion
  /* TimeRangeRoot in TimeRangeSelector */
```

File: static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+
+type Props = Omit<
+  ComponentProps<typeof TimeRangeSelector>,
+  'defaultPeriod' | 'onChange'
+> & {
+  className?: string;
+  defaultPeriod?: ComponentProps<typeof TimeRangeSelector>['defaultPeriod'];
+  onChange?: ComponentProps<typeof TimeRangeSelector>['onChange'];
```
Comment (priscilawebdev) [inline static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx:16]:
```suggestion
  defaultPeriod?: ComponentProps<typeof TimeRangeSelector>['defaultPeriod'];
  onChange?: ComponentProps<typeof TimeRangeSelector>['onChange'];
   className?: string;
```

🤷‍♀️

File: static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+function PageTimeRangeSelector({className, ...props}: Props) {
+  const [isCalendarOpen, setIsCalendarOpen] = useState(false);
+
+  return (
+    <DropdownDate className={className} isCalendarOpen={isCalendarOpen}>
+      <TimeRangeSelector
+        label={<DropdownLabel>{t('Date Range:')}</DropdownLabel>}
+        onToggleSelector={isOpen => setIsCalendarOpen(isOpen)}
+        defaultPeriod={DEFAULT_STATS_PERIOD}
+        onChange={() => {}}
```
Comment (priscilawebdev) [inline static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx:28]: should we update the `TimeRangeSelector` component, making this prop optional?

Comment (matejminar) [reply static/app/components/organizations/timeRangeSelector/pageTimeRangeSelector.tsx:28]: done!

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**